tools/network/bootstrap.go

```diff
@@ -37,35 +35,28 @@ func ReadFromSRV(service string, protocol string, name string, fallbackDNSResolv
 		return
 	}
 
-	var records []*net.SRV
-	if secure {
-		var dnssecError error
-		_, records, dnssecError = dnssec.LookupSRV(service, protocol, name)
-		if dnssecError != nil {
-			err = fmt.Errorf("ReadFromBootstrap: Failed to obtain SRV with DNSSEC: %s", dnssecError.Error())
-			return
+	controller := NewResolveController(secure, fallbackDNSResolverAddress)
+
+	systemResolver := controller.SystemResolver()
+	_, records, sysLookupErr := systemResolver.LookupSRV(context.Background(), service, protocol, name)
+	if sysLookupErr != nil {
+		log.Infof("ReadFromBootstrap: DNS LookupSRV failed when using system resolver: %v", sysLookupErr)
+
+		var fallbackLookupErr error
+		if fallbackDNSResolverAddress != "" {
+			fallbackResolver := controller.FallbackResolver()
+			_, records, fallbackLookupErr = fallbackResolver.LookupSRV(context.Background(), service, protocol, name)
 		}
-	} else {
-		var sysLookupErr error
-		_, records, sysLookupErr = net.LookupSRV(service, protocol, name)
-		if sysLookupErr != nil {
-			var resolver Resolver
-			// try to resolve the address. If it's an dotted-numbers format, it would return that right away.
-			// if it's a named address, we would attempt to parse it and might fail.
-			// ( failing isn't that bad; the resolver would internally try to use 8.8.8.8 instead )
-			if DNSIPAddr, err2 := net.ResolveIPAddr("ip", fallbackDNSResolverAddress); err2 == nil {
-				resolver.DNSAddress = *DNSIPAddr
-			} else {
-				log.Infof("ReadFromBootstrap: Failed to resolve fallback DNS resolver address '%s': %v; falling back to default fallback resolver address", fallbackDNSResolverAddress, err2)
-			}
+		if fallbackLookupErr != nil || fallbackDNSResolverAddress == "" {
```

Comment:
this doesn't align with the previous : If the ` fallbackLookupErr != nil`, it's already guarantee that `fallbackDNSResolverAddress != ""`. Also, please add the `fallbackDNSResolverAddress` to the info message.

Reply:
Old implementation internally used `resolver.DNSAddress` if set (fallback) or its default resolver otherwise.
New implementation explicitly checks for fallback address and uses it if set, or continues with default servers otherwise. So a new implementation will not fail if fallback is set to a dead server.